File: sample/src/main/java/com/codekul/sample/MainActivity.java
package com.codekul.sample;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void onCar(View view) {
        Toast.makeText(this, "Car", Toast.LENGTH_SHORT).show();
        ImageView img = findViewById(R.id.imgVeh);
        img.setImageResource(R.drawable.ic_directions_bike_black_24dp);
    }

    public void onPlane(View view) {
        Toast.makeText(this, "Plane", Toast.LENGTH_SHORT).show();
        ImageView img = findViewById(R.id.imgVeh);
        img.setImageResource(R.drawable.ic_chopper);
    }
}

File: asyncops/src/main/java/com/codekul/asyncops/MainActivity.kt
package com.codekul.asyncops

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.doAsync
import org.jetbrains.anko.indeterminateProgressDialog

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        btnOkay.setOnClickListener {
            val dlg = indeterminateProgressDialog (
                    message = "Wait a bit",
                    title = "Counting"
            )
            doAsync {
                for (i in 1..100) {
                    Thread.sleep(100)
                    runOnUiThread {
                        txtInfo.text = """$i"""
                    }
                }
                runOnUiThread { dlg.dismiss() }
            }
        }
    }

    private fun hndlrs() {
        val hndl = Handler(Looper.getMainLooper())
        Thread {
            for (i in 1..100) {
                Thread.sleep(1000)
                //txtInfo.text = """$i"""
                hndl.post {
                    txtInfo.text = """ $i """
                }
            }
        }.start()
    }

    private fun normal() {
        Thread {
            for (i in 1..100) {
                Thread.sleep(1000)
                //txtInfo.text = """$i"""
            }
        }.start()
    }
}
